tests/ordering/tests.py

```diff
@@ -307,3 +307,20 @@ def test_order_by_f_expression_duplicates(self):
             ],
             attrgetter("headline")
         )
+
+    def test_related_ordering_duplicate_table_reference(self):
+        """
+        An ordering referencing a model with an ordering referencing a model
+        multiple time no circular reference should be detected (#24654).
+        """
+        first_author = Author.objects.create(pk=1)
+        second_author = Author.objects.create(pk=2)
+        self.a1.author = first_author
+        self.a1.second_author = second_author
+        self.a1.save(update_fields={'author', 'second_author'})
```

Comment:
Probably, I don't think the benefit is worth the cognitive load of someone looking at the test and wondering about it.